tests/plugins/lti_consumer/test_models.py

```diff
@@ -5,190 +5,181 @@
-        instance.save()
-        auth_headers = {
-            "Authorization": (
-                'OAuth oauth_nonce="80966668944732164491378916897", oauth_timestamp="1378916897", '
-                'oauth_version="1.0", oauth_signature_method="HMAC-SHA1", '
-                'oauth_consumer_key="InsecureOauthConsumerKey", '
-                'oauth_signature="frVp4JuvT1mVXlxktiAUjQ7%2F1cw%3D"'
-            )
-        }
         expected_content_parameters = {
-            "lti_message_type": instance.lti_provider.get("display_name"),
+            "lti_message_type": "basic-lti-launch-request",
             "lti_version": "LTI-1p0",
-            "resource_link_id": str(instance.id),
+            "resource_link_id": "1234",
             "context_id": "example.com",
             "user_id": "richie",
             "lis_person_contact_email_primary": "",
-            "roles": "instructor",
-            "oauth_consumer_key": "InsecureOauthConsumerKey",
+            "roles": "student",
+            "oauth_consumer_key": "ManualTestOauthConsumerKey",
+            "oauth_nonce": "59474787080480293391616018589",
+            "oauth_signature": "nh2VyyxvKNTsEFIa68yFmWC+10w=",
             "oauth_signature_method": "HMAC-SHA1",
-            "oauth_timestamp": "1378916897",
-            "oauth_nonce": "80966668944732164491378916897",
+            "oauth_timestamp": "1616018589",
             "oauth_version": "1.0",
-            "oauth_signature": "frVp4JuvT1mVXlxktiAUjQ7/1cw=",
         }
+
         self.assertDictEqual(
-            expected_content_parameters,
-            instance.build_content_parameters(auth_headers, edit=True),
+            expected_content_parameters, instance.get_content_parameters()
         )
 
-    @mock.patch(
-        "richie.plugins.lti_consumer.models.LTIConsumer.authorize",
-        return_value={
-            "Authorization": (
-                'OAuth oauth_nonce="80966668944732164491378916897", oauth_timestamp="1378916897", '
-                'oauth_version="1.0", oauth_signature_method="HMAC-SHA1", '
-                'oauth_consumer_key="InsecureOauthConsumerKey", '
-                'oauth_signature="frVp4JuvT1mVXlxktiAUjQ7%2F1cw%3D"'
-            )
-        },
-    )
-    def test_models_lti_consumer_content_parameters(self, _):
+    @override_settings(RICHIE_LTI_PROVIDERS=get_lti_settings())
+    @mock.patch.object(LTIConsumer, "get_resource_link_id", return_value="1234")
+    def test_models_lti_consumer_auth_parameters_edit_mode(
+        self, _mock_rl, _mock_ts, _mock_nonce
+    ):
         """
-        Verify that LTI content consumption parameters are correctly built through the
-        content_parameters wrapper
+        Verify that instructor LTI authentication parameters are correctly built with edit mode
         """
-        instance = LTIConsumer(lti_provider_id="lti_provider_test")
-        instance.save()
+        instance = LTIConsumerFactory(
+            lti_provider_id="lti_provider_test",
```

Comment:
maybe we could do the same test for an instructor with no lti_provider_id wdyt ?

Reply:
Good idea! I was lazy ^^